webpack/components/RolesIndex/RolesIndex.js

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import { Table, TableHeader, TableBody } from '@patternfly/react-table';
+import { Pagination } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { linkCellFormatter, DEFAULT_PER_PAGE } from './RolesIndexConstants';
+
+const RolesIndex = ({ rowsData, RolesIndexactions }) => {
+  const cols = [
+    { title: 'Name' },
+    { title: 'Variables', cellTransforms: [linkCellFormatter] },
+    { title: 'Hostgroups' },
+    { title: 'Hosts', cellTransforms: [linkCellFormatter] },
+    { title: 'Imported at' },
+  ];
+  const rows = rowsData;
+  const [page, setPage] = useState();
+  const [perPage, setPerPage] = useState(DEFAULT_PER_PAGE);
+  const [paginatedRows, setPaginatedRows] = useState(rows.slice(0, perPage));
+
+  const handlePerPageSelect = (
+    event,
+    newPerPage,
+    newPage,
+    startIdx,
+    endIdx
+  ) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const handleSetPage = (event, newPage) => {
+    const startIdx = (newPage - 1) * perPage;
+    const endIdx =
+      rows.length < newPage * perPage ? rows.length : newPage * perPage;
+    setPage(newPage);
+    setPaginatedRows(rows.slice(startIdx, endIdx));
+  };
+
+  const renderPagination = (variant = 'top') => (
```

Comment:
It might be worth extracting to a separate component now that we is it in multiple [places](https://github.com/theforeman/foreman_ansible/blob/master/webpack/components/AnsibleRolesAndVariables/AnsibleRolesAndVariables.js#L91)

Reply:
Pagination is a component (of pattern fly) , here is just use if it, I'm not sure it's needed to extract to another component,

Reply:
Right, maybe it is not worth it.